### DISCUSSION TITLE
CMJ和ESM

### DISCUSSION THREAD

**Discussion by giscus** [ztkuaikuai/blog#7] [General]
# CMJ和ESM

记录一些技术、生活、阅读的感悟；是我的人类使用说明书。

https://blog.kuaikuaitz.top/posts/CMJ%E5%92%8CESM

**Comment by ztkuaikuai** [comment]
**由于CMJ是动态依赖，所以在打包时如果遇到CMJ，需要引入runtime去处理CMJ，来保证CMJ的语义是正确的。**

Rollup对CMJ的处理没有引入runtime，导致其在打包时可能会产生问题。

由于Vite现阶段生产使用Rollup，开发使用esbuild进行预构建，所以可能会导致开发和生产不一致的问题。
